pilot/pkg/config/kube/ingress/conversion.go

```diff
@@ -216,20 +216,22 @@ func ConvertIngressVirtualService(ingress v1beta1.Ingress, domainSuffix string,
 		if f {
 			vs := old.Spec.(*networking.VirtualService)
 			vs.Http = append(vs.Http, httpRoutes...)
-			sort.SliceStable(vs.Http, func(i, j int) bool {
```

Comment:
And I'm a little confused about the old behavior, it seems to wanna sort the routes, but only when merging, so if there's only one ingress the generated vs routes will not be sorted, is that right?

Reply:
I think so. I think the idea was to ensure if we have 2 ingress with something like:
```
match0: /bar
match1: /
```

and

```
match0: /foo
match1: /
```

we put the `/` matches at the bottom so /foo and /bar can match